### ISSUE TITLE
Installer hash does not match; this cannot be overridden when running as admin

### ISSUE THREAD

**Issue by ivonenchev** [microsoft/winget-pkgs#147885]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

Installer hash does not match; this cannot be overridden when running as admin

### Steps to reproduce

When I have tried to install the following packages : 

winget install --id=Google.Chrome -e; winget install --id=Ubisoft.Connect -e;  winget install --id=Viber.Viber -e; winget install --id=ElectronicArts.EADesktop -e; winget install --id=GOMLab.GOMPlayer

### Actual behavior

Installer hash does not match; this cannot be overridden when running as admin

### Expected behavior

To be installed successfully  

### Environment

```raw
PS C:\Users> winget --info
Windows Package Manager v1.7.10861
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22631.3374
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.22.10861.0

Winget Directories
-------------------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
PS C:\Users>
```

**Comment by Dragon1573** [comment]
Validate these manifests in GitHub Workflows again:

- ✅ _Google Chrome_: See [logs](https://github.com/Dragon1573/winget-test/actions/runs/8570977736/job/23490179810#step:5:26)
- ⚠️ _Ubisoft Connect_: See [logs](https://github.com/Dragon1573/winget-test/actions/runs/8570989270/job/23490218280#step:5:27)
- ⚠️ _Viber_: See [logs](https://github.com/Dragon1573/winget-test/actions/runs/8571140721/job/23490721191#step:5:25)
- ⚠️ _EA Desktop_: See [logs](https://github.com/Dragon1573/winget-test/actions/runs/8571177468/job/23490843539#step:5:21)
- ⚠️ _GOMPlayer_: See [logs](https://github.com/Dragon1573/winget-test/actions/runs/8571230499/job/23491009833#step:5:21)

-----

It seems all above 5 packages are using vanity URL. Do they provide correct hash on their download site?

**Comment by Kaligula0** [comment]
possible duplicate of yours #137039 ?

**Comment by Trenly** [comment]
> possible duplicate of yours #137039 ?

Similar packages, but not entirely duplicate. I'll submit the updates which should resolve them both though

**Comment by Kaligula0** [comment]
> Similar packages, but not entirely duplicate. I'll submit the updates which should resolve them both though

Yeah, I just noticed that you've struggled with the ElectronicArts.EADesktop error since January :)

**Comment by remidebette** [comment]
Hi, still getting this, isn't the PR merged?
Thank you

**Comment by stephengillie** [comment]
> Hi, still getting this, isn't the PR merged? Thank you

Hi @remidebette,

I'm guessing that the issue hasn't relapsed, but reoccurred. The `InstallerUrl` is what we frequently call a "vanity URL" as it's used more often for marketing and promotions than for archival and historical record - it always has the latest version of the software package. The problem is that it's newer than the version described in our repo. The fix is the same as before - have a PR submitted and approved to update the manifest with the new hash. We have an automated system that checks some of these vanity URLs every day, but unfortunately not all of them.

**Comment by remidebette** [comment]
Thanks for the response.
In the case of EADesktop, or Discord, my understanding is that the apps are evergreen and automate without an issue.
Couldn't there be a parameter in the manifest to remove the need to upgrade those through winget like brew does?

**Comment by Trenly** [comment]
@remidebette - The main issue is that the self-updaters for these apps don't write the new version information to the registry. This means if you installed version 13.166.0.5679, and it later auto-updated itself to 13.167.0.6312, all applications within windows (Control Panel, Apps & Features in Settings, WinGet. . .)  will still detect it as 13.166.0.5679.

There isn’t a way to fix this in the manifest - since the manifest simply contains information about a version. The only way to truly fix this is for the publishers of the applications to fix their self-updaters to write the newly installed version into the registry

What you can do, though, is run `winget pin add <id>` on your machine to pin the application. This will exclude it from `winget upgrade --all` by default

**Comment by stephengillie** [comment]
What would be needed is basically an `UpgradeBehavior` named something like `UpgradesSelf`, which would tell the package manager to usually skip this package unless there were an issue or override. But that doesn't currently exist, and would need to be a feature request. 

The closest that currently exists would be `UpgradeBehavior: deny`

> If the package should not be upgraded through WinGet, the "deny" value should be specified.
- https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.6.0/installer.md (under UpgradeBehavior)

But `deny` might be heavy-handed for auto-updating applications, as it seems like it would prevent all updating.  Unless it has an override option?

**Comment by Trenly** [comment]
> What would be needed is basically an `UpgradeBehavior` named something like `UpgradesSelf`, which would tell the package manager to usually skip this package unless there were an issue or override. But that doesn't currently exist, and would need to be a feature request.
> 
> The closest that currently exists would be `UpgradeBehavior: deny`
> 
> > If the package should not be upgraded through WinGet, the "deny" value should be specified.
> 
> * https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.6.0/installer.md (under UpgradeBehavior)
> 
> But `deny` might be heavy-handed for auto-updating applications, as it seems like it would prevent all updating. Unless it has an override option?

`winget install --force` will override the `deny` behavior

**Comment by stephengillie** [comment]
Should `UpgradeBehavior: deny` be used for self-updating applications whose updater doesn't also update the Registry?

**Comment by Trenly** [comment]
> Should `UpgradeBehavior: deny` be used for self-updating applications whose updater doesn't also update the Registry?

It depends. Usually, `RequireExplicitUpgrade` is enough for that case. Discord has it set to deny because using the installer to update the existing installation will wipe out some usernsettings

**Comment by ivonenchev** [comment]
Fixed.
